src/Repo_new.java

```diff
@@ -0,0 +1,804 @@
+            }
+
+            f = StarTeamFinder.findFile(folder, name, true);
+
+            if (f != null) {
+                _fileCache.put(path, f);
+            }
+        }
+
+        return f;
+    }
+
+    public synchronized Folder getOrCreateFolder(final String path) {
+        Folder f = _folderCache.get(path);
+
+        if (f == null) {
+            final Folder root = getRootFolder();
+            f = StarTeamFinder.findFolder(root, path);
+
+            if (f == null) {
+                final int ind = path.lastIndexOf('/');
+
+                if (ind != -1) {
+                    f = getOrCreateFolder(path.substring(0, ind));
+                    f = new Folder(f);
+                    f.setName(path.substring(ind + 1));
+                    f.update();
+                } else {
+                    f = new Folder(root);
+                    f.setName(path);
+                    f.update();
+                }
+            }
+
+            _folderCache.put(path, f);
+        }
+
+        return f;
+    }
+
+    public synchronized com.starbase.starteam.File getOrCreateFile(
+            final String path) {
+        com.starbase.starteam.File f = _fileCache.get(path);
+
+        if (f == null) {
+            final int slash = path.lastIndexOf('/');
+            final String name;
+            final Folder folder;
+
+            if (slash == -1) {
+                name = path;
+                folder = getRootFolder();
+            } else {
+                name = path.substring(slash + 1);
+                folder = getOrCreateFolder(path.substring(0, slash));
+            }
+
+            f = StarTeamFinder.findFile(folder, name, true);
+
+            if (f == null) {
+                f = new com.starbase.starteam.File(folder);
+                f.setName(name);
+            }
+
+            _fileCache.put(path, f);
+        }
+
+        return f;
+    }
+
+    public synchronized void rename(final Item i, final String dest,
+            final String comment) {
+        final String oldPath = getPath(i);
+        i.setComment(comment);
+        i.moveTo(getOrCreateParentFolder(dest));
+
+        if (i instanceof com.starbase.starteam.File) {
+            final com.starbase.starteam.File f = (com.starbase.starteam.File) i;
+            f.setName(Repo.getFileName(dest));
+            f.update();
+            _fileCache.remove(oldPath);
+            _fileCache.put(getPath(f), f);
+        } else {
+            final Folder f = (Folder) i;
+            f.setName(Repo.getFileName(dest));
+            f.update();
+            _folderCache.remove(oldPath);
+            _folderCache.put(getPath(f), f);
+        }
+    }
+
+    public MessageFormat getCommentFormat() {
+        return _commentFormat;
+    }
+
+    public boolean isGitStComment(final String comment) {
+        return comment.endsWith(FOOTER_END) && comment.contains(FOOTER_START);
+    }
+
+    public String getPrevImportSha() throws InterruptedException, IOException {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to throw at most one checked exception instead of: java.lang.InterruptedException, java.io.IOException [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=squid%3AS1160)